lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.hmpps.sqs
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "hmpps.sqs")
+data class HmppsSqsProperties(
+  val provider: String = "aws",
+  val region: String = "eu-west-2",
+  val localstackUrl: String = "http://localhost:4566",
+  val queues: Map<String, QueueConfig>,
+) {
+  data class QueueConfig(
+    val queueName: String,
+    val queueAccessKeyId: String = "",
+    val queueSecretAccessKey: String = "",
+    val asyncQueueClient: Boolean = false,
+    val dlqName: String,
+    val dlqAccessKeyId: String = "",
+    val dlqSecretAccessKey: String = "",
+    val asyncDlqClient: Boolean = false,
+  )
+
+  init {
+    queues.forEach { (queueId, queueConfig) ->
+      if (queueId != queueId.lowercase()) throw InvalidHmppsSqsPropertiesException("queueId $queueId is not lowercase")
```

Comment:
I discovered that Spring cannot unwrap environment variables of the format `HMPPS_SQS_QUEUES_QUEUE_NAME_QUEUE_ACCESS_KEY_ID" into the queues Map in this class, so the queueId has to be lower case.